### DISCUSSION TITLE
Get-PnPPage returns Object reference not set to an instance of an object.

### DISCUSSION THREAD

**Discussion by ghost** [pnp/powershell#1577] [General]
### Reporting an Issue or Missing Feature
Issue

### Expected behavior 
Should return Page object

### Actual behavior 
Throws exception

### Steps to reproduce behavior
First, I connect to my site like so:

```
Connect-PnPOnline -Url https://mycompany.sharepoint.com/product/development/repos/ -UseWebLogin
```

I can run `Get-PnPFolderItem -Identity "Pages" and see:

```
Name          Type   Items/Size Last Modified
----          ----   ---------- -------------
Forms         Folder          0 1/26/2022 1:50:02 AM
AllRepos.aspx File       902349 1/26/2022 11:20:17 PM
default.aspx  File         4348 1/26/2022 5:34:27 PM
NewRepos.aspx File       481474 1/26/2022 11:24:48 PM
```

Next, I try to get one of the sites with 

```
Get-PnPPage -Identity "default.aspx"
```

I get the following exception:

```
Get-PnPPage : Object reference not set to an instance of an object.
At line:1 char:1
+ Get-PnPPage -Identity "default.aspx"
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Get-PnPPage], NullReferenceException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Pages.GetPage
```

The `Get-PnPException` command returns:

```
Message          : Object reference not set to an instance of an object.
Stacktrace       :    at PnP.Core.Model.SharePoint.Page.<GetPagesListData>d__99.MoveNext()
                   --- End of stack trace from previous location where exception was thrown ---
                      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                      at PnP.Core.Model.SharePoint.Page.<LoadPagesAsync>d__98.MoveNext()
                   --- End of stack trace from previous location where exception was thrown ---
                      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                      at PnP.Core.Model.SharePoint.Web.<GetPagesAsync>d__288.MoveNext()
                   --- End of stack trace from previous location where exception was thrown ---
                      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                      at PnP.Core.Model.SharePoint.Web.GetPages(String pageName)
                      at PnP.PowerShell.Commands.Base.PipeBinds.PagePipeBind.GetPage()
                      at PnP.PowerShell.Commands.Pages.GetPage.ExecuteCmdlet()
                      at PnP.PowerShell.Commands.PnPSharePointCmdlet.ProcessRecord()
ScriptLineNumber : 1
```

Note, I can put anything I want for the identity:

```
Get-PnPPage -Identity "BlahBlah"
```

and I get the same exception, so I don't think it's even getting to the part of looking up the site.

I think this might be due to the way I connect to the site, but I'm not sure of another way to do it.  I think either Get-PnPPage doesn't support subsites (or nested subsites?) or it doesn't support whatever auth method I'm using.

### What is the version of the Cmdlet module you are running?
1.9.0

### Which operating system/environment are you running PnP PowerShell on?
Windows

**Comment by ghost** [comment]
Update on this.  It's something to do with my subsite.  If I connect to one level below, `https://mycompany.sharepoint.com/product/development/`, I can get pages just fine.

**Comment by gautamdsheth** [comment]
Yeah you need to connect to the site or subsite where pages reside to retrieve them.
Closing the issue.

**Comment by ghost** [comment]
@gautamdsheth I don't understand why this was closed, what's the work around for the issue?  I need to be able to access pages on the /product/development/repos/ site.  If I connect to that site, I get an exception.  I can connect to and access pages on the parent site just fine.  Can you provide me with a way to connect to /development/repos and access pages on /development/repos/ without getting the exception?  Is there anything I need to configure on the site for this to work?  I'm an owner of the site so I don't think it's permissions.  Why does the cmdlet work on one site, but not another?  Sorry if my question doesn't make sense.  Please feel free to ask me more questions instead of closing the bug.  We really need to get this working.

**Comment by veronicageek** [comment]
@mike-christensen - No issue here getting pages on subsites

![image](https://user-images.githubusercontent.com/25181757/151425809-f6e6f177-96f5-4cd9-86f6-54de8ee454bf.png)

**Comment by ghost** [comment]
@veronicageek Yea, it must be something to do with the actual site since I can connect and get pages on some subsites, but not others..  The "Object reference not set to an instance of an object." exception is quite vague.  Is there a way to get more details about the problem?

**Comment by gautamdsheth** [comment]
Ok I assumed that the issue is fixed because you mentioned it yourself that you connected to subsite and it started working.

Also Get-PnPPage is used to get modern pages from site pages library. 

If you point it towards publishing pages or wiki pages , it will not work as it is not supported . This command is not intended to work with other libraries like Pages library and fetch publishing pages.

**Comment by ghost** [comment]
@gautamdsheth Ok that makes sense.  My end goal is to create and publish my own HTML page, but have it have the site master page and menus/navigation.  I can publish a "document", which I've gotten to work, but then I'm responsible for 100% of the HTML.  Basically, we're looking at Sharepoint as a way to host some content we create using a script that generates a bunch of HTML..  We could probably use an App Service or something, but then we'd have to implement authentication and all that boring stuff.

I've posted [this question](https://sharepoint.stackexchange.com/questions/300109/creating-a-brand-new-sharepoint-page-using-powershell-add-pnpfile-command) on StackExchange in hopes of getting someone to help me.  Maybe you can point me in the right direction, since it seems that the "PnPPage" Cmdlets might not be what I'm looking for.  Thanks!

**Comment by gautamdsheth** [comment]
Moving this to a discussion
